File: packages/driver-interface/services/pelias.js
const fetch = require('node-fetch')

const getAddressFromCoordinate = async ({ lon, lat }) =>
  fetch(
    `https://pelias.iteamdev.io/v1/reverse?point.lat=${lat}&point.lon=${lon}`
  )
    .then((res) => res.json())
    .then(({ features }) => features[0].properties.label)

module.exports = {
  getAddressFromCoordinate,
}

File: packages/engine_umbrella/apps/message_generator/lib/adapters/pelias.ex
defmodule MessageGenerator.Adapters.Pelias do
  def get_address_info_from_coordinates(lat, lon) do
    HTTPoison.get!("https://pelias.iteamdev.io/v1/reverse?point.lat=#{lat}&point.lon=#{lon}")
    |> Map.get(:body)
    |> Jason.decode!()
    |> Map.get("features")
    |> List.first()
    |> Map.get("properties")
    |> Map.take(["name", "county", "street"])
  end
end
